internal/plumbing/tree_diff.go

```diff
@@ -20,6 +21,7 @@ import (
 type TreeDiff struct {
 	core.NoopMerger
 	SkipDirs     []string
+	Filter       string
```

Comment:
Better to rename this to `NameFilter` to avoid confusion; also, since this is a regular expression, we can change the type to `*regexp.Regexp` and `MustCompile()` it in the configuration.